digitalocean_k8s_ubuntu_16.04_node.sh

```diff
@@ -42,9 +58,23 @@ systemctl daemon-reload
 systemctl restart kubelet
 
 # Parse kubicorn configuration file.
-TOKEN=$(cat /etc/kubicorn/cluster.json | jq -r '.clusterAPI.spec.providerConfig' | jq -r '.values.itemMap.INJECTEDTOKEN')
-MASTER=$(cat /etc/kubicorn/cluster.json | jq -r '.clusterAPI.spec.providerConfig' | jq -r '.values.itemMap.INJECTEDMASTER')
+TOKEN=$(< /etc/kubicorn/cluster.json jq -r '.clusterAPI.spec.providerConfig' | jq -r '.values.itemMap.INJECTEDTOKEN')
+MASTER=$(< /etc/kubicorn/cluster.json jq -r '.clusterAPI.spec.providerConfig' | jq -r '.values.itemMap.INJECTEDMASTER')
 
 # Join node a cluster.
-kubeadm reset
-kubeadm join --node-name ${HOSTNAME} --token ${TOKEN} ${MASTER} --discovery-token-unsafe-skip-ca-verification
+kubeadm reset --force
+
+# Delay kubeadm join until master is ready
+attempts=0
+while [ $(( attempts++ )) -lt $MAX_ATTEMPTS ]; do
+  curl --silent --connect-timeout 10 -k "${MASTER}" && break
```

Comment:
What does the pinging master IP returns?

Would it make more sense to use `/healthz` here if it doesn't require providing cert and key files?

Reply:
I have checked and /healthz doesn't require auth, so I used this approach instead, only joining once /healthz returns a 200 response